nselib/msrpctypes.lua

```diff
@@ -4523,6 +4523,27 @@ function unmarshall_SERVICE_STATUS(data, pos)
 end
 
 
+--- Unmarshalls a null-terminated Unicode string (LPTSTR datatype)
+-- @param w_str The data being processed
+-- @param startpos  The current position within the data
+-- @return The new position
+-- @return The unmarshalled string
+function unmarshall_lptstr(w_str, startpos)
+
+  local endpos = startpos
+  local length = w_str:len()
+  local s = ""
+
+  while ( s ~= "\0\0" and endpos < length ) do
+    -- Reads the next character.
+    s = string.sub(w_str, endpos, endpos + 1)
+
+    -- We add 2 because we are dealing with UTF-16 format.
+    endpos = endpos + 2
+  end
+
+  return endpos, string.sub(w_str, startpos, endpos)
```

Comment:
You are not differentiating between "whatever\0\0" and "whatever". Should not you fail in the latter case?

Also, make sure that you have the indices fed to `string.sub` correct. Test your code against string `"aabbc\0\0cddee\0\0ff"`.